pallets/task/src/tests.rs

```diff
@@ -423,19 +427,28 @@ fn task_can_be_removed_by_owner(){
 	new_test_ext().execute_with( || {
 
 		assert_ok!(Profile::create_profile(Origin::signed(1), username(), interests(), HOURS));
+		let signer_balance = Balances::balance(&1);
 
 		// Ensure new task can be created.
 		assert_ok!(Task::create_task(Origin::signed(1), title(), spec(), BUDGET, get_deadline(), attachments(), keywords()));
+		let signer_balance_after_task_creation = Balances::balance(&1);
 
 		// Ensure that task can't be started once its started
 		let hash = Task::tasks_owned(1)[0];
+		let task_balance = Balances::balance(&Task::account_id(&hash));
+		assert_eq!(task_balance, BUDGET);
+		assert_eq!(signer_balance, signer_balance_after_task_creation + task_balance);
 
 		// Ensure another user can't remove the task
 		assert_noop!(Task::remove_task(Origin::signed(2), hash), Error::<Test>::NoPermissionToRemove);
 
 		// Ensure the task can be removed
 		assert_ok!(Task::remove_task(Origin::signed(1), hash));
 		assert_eq!(Task::task_count(), 0);
+		let signer_balance_after_task_deletion = Balances::balance(&1);
+		let task_balance = Balances::balance(&Task::account_id(&hash));
+		assert_eq!(signer_balance, signer_balance_after_task_deletion);
+		assert_eq!(task_balance, 0);
```

Comment:
Nice test. 

Just please create new unit tests. 

Each tests should tests a small unit of code. I is better to create new tests when adding new functionality rather than put everything in one big test.